fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt

```diff
@@ -509,6 +509,47 @@ class HomeScreenRobot {
                 },
             )
     }
+    fun verifyNimbusMessageCard(title: String, text: String, action: String) {
+        // Checks message Title
+        assertTrue(
+            mDevice.findObject(
+                UiSelector()
+                    .className("androidx.compose.ui.platform.ComposeView")
+                    .className("android.view.View")
+                    .className("android.widget.TextView")
+                    .textContains(title),
+            ).waitForExists(waitingTime),
+        )
+        // Checks main message text
+        assertTrue(
+            mDevice.findObject(
+                UiSelector()
+                    .className("androidx.compose.ui.platform.ComposeView")
+                    .className("android.view.View")
+                    .className("android.widget.TextView")
```

Comment:
Style: I feel sure there are higher level primitives that this, e.g. `onView(…).check(…)`, or 

```kotlin
        val textView = UiSelector()
            .className(ComposeView::class.java)
            .className(View::class.java)
            .className(TextView::class.java)
        assertTrue(
            mDevice.findObject(textView.textContains(title)).waitForExists(waitingTime)
        )
        assertTrue(
            mDevice.findObject(textView.textContains(text)).waitForExists(waitingTime)
        )
        assertTrue(
            mDevice.findObject(textView.textContains(action)).waitForExists(waitingTime)
        )
```

Reply:
Oh this is cool, what is this called? Specifically ```View::class.java```

Reply:
The `View::class.java` is called a Class literal. Each type of object is called a class.

The `UiSelector` is a query; the `textView` query is looking for a `TextView` inside a `View` inside a `ComposeView`.

Each of the `assertTrue` statements extends the `textView` query to make queries for text containing a title, text and action.

The `findObject` actually executes the queries.